### ISSUE TITLE
Performance enhancement for ML-KEM

### ISSUE THREAD

**Issue by baentsch** [openssl/openssl#25879]
This is to suggest work (for now in the ml-kem feature branch and of course only after https://github.com/openssl/openssl/pull/25848 landed) to optimize performance of the overall ML-KEM logic.

This may entail
- [ ] better provider integration as per https://github.com/openssl/openssl/pull/25848#discussion_r1829211531
- [ ] optimized code for different architectures
- [ ] a general code re-write
- [ ] (your proposal here)

**Comment by t8m** [comment]
IMO this is not necessary for MVP (3.5). Of course it would be nice to have. I assume the main performance gains would be from assembler optimizations.

**Comment by vdukhovni** [comment]
I think that `better provider integration` is no longer pertinent. Or at least needs to be more clear as what's intended.
Similarly, not clears what `general code re-write` is about.

**Comment by baentsch** [comment]
> I think that `better provider integration` is no longer pertinent. Or at least needs to be more clear as what's intended. Similarly, not clears what `general code re-write` is about.

This is a pretty old issue listing many things that had not been done then. Main focus by now would be assembly optimizations for the core logic and the key platforms. Text changed accordingly.
